TOOLS/ce/src/main/scala/org.opalj.ce/ConfigObject.scala

```diff
@@ -0,0 +1,208 @@
+     * This means that the objects are free of conflicting values and lists. Objects are allowed to overlap as long as there are no conflicts down the tree.
+     * @param insertingObject Is the object that is supposed to be merged into the executing one.
+     */
+    def merge(insertingObject: ConfigObject): Unit = {
+
+        // Expanding both objects guarantees compatible key naming syntax
+        expand()
+        insertingObject.expand()
+
+        // Insert object
+        for (kvpair <- insertingObject.entries) {
+            val (key, value) = kvpair
+            if (entries.contains(key)) {
+                val conflicting_entry = entries.getOrElse(key, null)
+                if (conflicting_entry.isInstanceOf[ConfigObject] && value.isInstanceOf[ConfigObject]) {
+                    val conflicting_child_object = conflicting_entry.asInstanceOf[ConfigObject]
+                    conflicting_child_object.merge(value.asInstanceOf[ConfigObject])
+                } else {
+                    OPALLogger.error("Configuration Explorer", s"Info on incompatible keys: ${key.trim}")
+                    throw new IllegalArgumentException(
+                        s"Unable to merge incompatible types: ${value.getClass} & ${conflicting_entry.getClass}"
+                    )
+                }
+            } else {
+                OPALLogger.info("Configuration Explorer", s"No conflict detected. Inserting ${key.trim}")
+                entries += kvpair
+            }
+        }
+
+        collapse()
+    }
+
+    /**
+     * This method collapses the object structure by joining inheriting objects containing only one value.
+     * Inverse function of expand.
+     */
+    def collapse(): Unit = {
+        for (entry <- entries) {
+            val (key, value) = entry
+            value.collapse()
+
+            // If the entry is a config object with exactly one child -> merge
+            if (value.isInstanceOf[ConfigObject]) {
+                val value_object = value.asInstanceOf[ConfigObject]
+                if (value_object.entries.size == 1) {
+                    // Merge Keys
+                    val (childkey, childvalue) = value_object.entries.head
+                    val newkey = key.trim + "." + childkey.trim
+
+                    // Merge comments
+                    childvalue.comment = childvalue.comment.mergeComment(value_object.comment)
+
+                    // Add new object
+                    entries += (newkey -> childvalue)
+
+                    // Remove old object
+                    entries -= key
+                }
+            }
+        }
+        if (entries.size == 1) {
+            if (comment.isEmpty) {} else {
+                val (key, value) = entries.head
+                if (value.comment.isEmpty) {}
+            }
+        }
+    }
+
+    /**
+     * This method expands the current object to represent all objects within the structure.
+     * Inverse function of collapse.
```

Comment:
Not exactly true. `collapse` merges comments that are not split again by `expand`, so this is not isomorph